doc/index.rst

```diff
@@ -205,13 +205,13 @@
                     <h4>News</h4>
                     <ul>
                     <li><em>On-going development:</em>
-                    <a href="whats_new.html"><em>What's new</em> (Changelog)</a>
+                    <a href="/dev/whats_new.html"><em>What's new</em> (Changelog)</a>
```

Comment:
I think you are right actually, `/dev/whats_new.html` does work, not sure how I tested this orginally.

The CircleCI build looks fine (although it doesn't test the `/dev/whats_new.html`). +1 for merge.